src/pathpyG/core/Graph.py

```diff
@@ -62,23 +63,15 @@ def __init__(self, edge_index: torch.Tensor,
                                     edge_weight=torch.FloatTensor([1.0, 2.5, 0.7[]))
             ```
         """
-        if node_id is None:
-            node_id = []
+        if mapping is None:
+            self.mapping = IndexMap()
+        else:
+            self.mapping = mapping
 
-        assert len(node_id) == len(set(node_id)), "node_id entries must be unique"
 
         # Create pyG Data object
-        if len(node_id) == 0:
-            num_nodes = max(max(edge_index[0]).item(), max(edge_index[1]).item())+1
-            self.data = Data(edge_index=edge_index, node_id=node_id, num_nodes = num_nodes, **kwargs)
-        else:
-            self.data = Data(
-                edge_index=edge_index, node_id=node_id, num_nodes=len(node_id), **kwargs
-            )
-
-        # create mappings between node ids and node indices
-        self.node_index_to_id = dict(enumerate(node_id))
-        self.node_id_to_index = {v: i for i, v in enumerate(node_id)}
+        num_nodes = int(max(max(edge_index[0]).item(), max(edge_index[1]).item())+1)
```

Comment:
Sure it does :-) Changed in latest commit.

However, we anyway need to rethink the semantics of num_nodes in pathpyG. Rght now, we enforce the pyG semantics, i.e. if a node index is missing in the edge_index then the node is assumed to exist as an isolated node. So a graph created from an edge_index [[3,1],[1,3]] is assumed to have four nodes and two edges, where the nodes with indices 0 and 2 are isolated. This is a design choice in pyG but it creates some unintuitive downstream effects, e.g. when creating time-aggregated networks from rolling windows ...

We may thus want to implement a remapping feature, so that the indices are remapped to contiguous indices, while adjusting the ID to index mapping accordingly.